aiida_quantumespresso/tools/pwinputparser.py

```diff
@@ -318,3 +321,96 @@ def parse_k_points(txt):
     return info_dict
 
 
+def create_builder_from_file(input_folder, input_file_name, code, metadata, pseudo_folder_path=None, use_first=False):
+    """Create a populated process builder for a Pw calculation,
+    from a standard QE input file and pseudo (upf) files
+
+    :param input_folder: the folder containing the input file
+    :type input_folder: aiida.common.folders.Folder or str
+    :param input_file_name: the name of the input file
+    :type input_file_name: str
+    :param code: the code associated with the calculation
+    :type code: aiida.orm.Code or str
+    :param metadata: metadata values for the calculation (e.g. resources)
+    :type metadata: dict
+    :param pseudo_folder_path: the folder containing the upf files (if None, then input_folder is used)
+    :type pseudo_folder_path: aiida.common.folders.Folder or str or None
+    :param use_first: parsed to UpfData.get_or_create
+    :type use_first: bool
+    :raises NotImplementedError: if the structure is not ibrav=0
+    :return: a builder instance for PwCalculation
+
+    """
+    pw_calc_cls = CalculationFactory('quantumespresso.pw')
+
+    inputs = {}
+    inputs["metadata"] = metadata
```

Comment:
Minor nitpick, I am trying to maintain the standard of using single quotes for identifier strings, such as keys in dictionaries